src/edu/ucsb/nceas/metacat/doi/datacite/DataCiteMetadataFactory.java

```diff
@@ -0,0 +1,556 @@
+    public static final String OPEN_IDENTIFIER = "<identifier identifierType=\"DOI\">";
+    public static final String CLOSE_IDENTIFIER = "</identifier>";
+    public static final String OPEN_CREATORS = "<creators>";
+    public static final String CLOSE_CREATORS = "</creators>";
+    public static final String OPEN_CREATOR = "<creator>";
+    public static final String CLOSE_CREATOR = "</creator>";
+    public static final String OPEN_CREATORNAME = "<creatorName>";
+    public static final String CLOSE_CREATORNAME = "</creatorName>";
+    public static final String OPEN_NAMEIDENTIFIER = "<nameIdentifier schemeURI=\"http://orcid.org/\" nameIdentifierScheme=\"ORCID\">";
+    public static final String CLOSE_NAMEIDENTIFIER = "</nameIdentifier>";
+    public static final String OPEN_AFFILICATION = "<affiliation>";
+    public static final String CLOSE_AFFILICATION = "/affiliation>";
+    public static final String OPEN_TITLES = "<titles>";
+    public static final String CLOSE_TITLES = "</titles>";
+    public static final String OPEN_TITLE_WITHLONG_ATTR = "<title xml:lang=\"";
+    public static final String CLOSE_TITLE= "</title>";
+    public static final String OPEN_PUBLISHER = "<publisher>";
+    public static final String CLOSE_PUBLISHER = "</publisher>";
+    public static final String OPEN_PUBLISHYEAR = "<publicationYear>";
+    public static final String CLOSE_PUBLISHYEAR = "</publicationYear>";
+    public static final String OPEN_SUBJECTS = "<subjects>";
+    public static final String CLOSE_SUBJECTS = "</subjects>";
+    public static final String OPEN_SUBJECT_WITHLONGATT = "<subject xml:lang=\"";
+    public static final String CLOSE_SUBJECT = "</subject>";
+    public static final String OPEN_LANGUAGE = "<language>";
+    public static final String CLOSE_LANGUAGE = "</language>";
+    public static final String OPEN_RESOURCETYPE_WITHTYPEGENERALATT = "<resourceType resourceTypeGeneral=\"";
+    public static final String CLOSE_RESROUCETYPE = "</resourceType>";
+    public static final String OPEN_FORMATS = "<formats>";
+    public static final String CLOSE_FORMATS = "</formats>";
+    public static final String OPEN_FORMAT = "<format>";
+    public static final String CLOSE_FORMAT = "</format>";        
+    public static final String OPEN_VERSION = "<version>";
+    public static final String CLOSE_VERSION = "</version>";
+    public static final String OPEN_DESCRIPTIONS = "<descriptions>";
+    public static final String CLOSE_DESCRIPTIONS = "</descriptions>";
+    public static final String OPEN_DESCRITPION_WITHLANGATT = "<description  descriptionType=\"Abstract\" xml:lang=\"";
+    public static final String CLOSE_DESCRIPTION = "</description>";
+    
+    public static final String CLOSE_ATT="\">";
+    public static final String EN = "en";
+    public static final String XML_LANG= "xml:lang";
+    
+    public static final String NAMESPACE = "http://datacite.org/schema/kernel-3";
```

Comment:
I'm wondering if we should upgrade our DataCite Metadata to Kernel version 4.1 since we are doing this refactor.  I can't imagine that the schema is that much different, but it may support more fields that we think are useful. @amoeba @taojing2002 @mbjones Thoughts?

Reply:
EZID library only supports version 3.1 now.

Reply:
I'd put in a strong 👍 	for publishing v4.1 since it's the latest and it's also about 2 years old at this point.